src/Discovery/DiscoveryActionBar/components/DownloadZipButton.tsx

```diff
@@ -3,89 +3,98 @@ import { DownloadOutlined } from '@ant-design/icons';
 import { Popover, Button, Modal } from 'antd';
 import handleDownloadZipClick from '../utils/handleDownloadZipClick';
 import { handleRedirectToLoginClickResumable } from '../../Utils/HandleRedirectToLoginClick';
+import isManifestDataMissing from '../../Utils/isManifestDataMissing';
+import { DiscoveryResource } from '../../Discovery';
 
 const DownloadZipButton = ({
   props, healIDPLoginNeeded, onlyInCommonMsg, downloadStatus, setDownloadStatus, history, location,
-}) => (
-  props.config.features.exportToWorkspace?.enableDownloadZip ? (
-    <React.Fragment>
-      <Popover
-        className='discovery-popover'
-        arrowPointAtCenter
-        content={(
-          <React.Fragment>
-            {healIDPLoginNeeded.length > 0
-              ? onlyInCommonMsg
-              : 'Directly download data (up to 250Mb) from selected studies'}
-          </React.Fragment>
-        )}
-      >
-        <Button
-          onClick={async () => {
-            if (props.user.username && !(healIDPLoginNeeded.length > 0)) {
-              handleDownloadZipClick(
-                props.config,
-                props.discovery.selectedResources,
-                downloadStatus,
-                setDownloadStatus,
-                history,
-                location,
-                healIDPLoginNeeded.length > 0,
-              );
-            } else {
-              handleRedirectToLoginClickResumable('download', props, history, location);
-            }
-          }}
-          type='default'
-          className={`discovery-action-bar-button${
-            props.discovery.selectedResources.length === 0 ? '--disabled' : ''
-          }`}
-          disabled={
-            props.discovery.selectedResources.length === 0
-            || downloadStatus.inProgress
-          }
-          icon={<DownloadOutlined />}
-          loading={downloadStatus.inProgress}
+}) => {
+  const checkIfDownloadZipButtonDisabled = () => {
+    const noSelectedResources = props.discovery.selectedResources.length === 0;
+    const downloadInProgress = downloadStatus.inProgress;
+    const eachSelectedResourcesIsMissingManifest = props.discovery.selectedResources.every(
+      (resource: DiscoveryResource) => isManifestDataMissing(resource),
+    );
```

Comment:
nitpicking: I think we should be able to make this function return early if any of these values evaluate to true, so it can skip checking the truthness of values in the next

Reply:
Both this function and the version of the function in DownloadManifestButton have been updated to return early if any of the values evaluate as true, otherwise they return ```false```